### ISSUE TITLE
Cannot login to /metric using credentials or API key

### ISSUE THREAD

**Issue by qnlbnsl** [louislam/uptime-kuma#3307]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

I was trying to connect home assistant to the /metrics endpoint but was unabel to login. When i tried manually it would not allow me to login at all. I tried a fresh image as well and it did not work.
I am running uptime kuma as a service in a docker swarm using portainer. Any and all communication is done via a reverse proxy.

### 👟 Reproduction steps

Go to <host>/metrics
basic auth popup from browser comes up
enter login details.

### 👀 Expected behavior

login is successful 

### 😓 Actual Behavior

login does nor succeed

### 🐻 Uptime-Kuma Version

1.21.3

### 💻 Operating System and Arch

Docker Image

### 🌐 Browser

Microsoft Edge with added security enabled

### 🐋 Docker Version

Docker version 24.0.2, build cb74dfc

### 🟩 NodeJS Version

Whatever is default in the docker image

### 📝 Relevant log output

```shell
That is the thing. No log is generated. I do see regular login attempts but nothing else.
```

**Comment by louislam** [comment]
Changed to `help` as considering a lot of users are already using it without problems.

Did you try Chrome/Firefox?

**Comment by chakflying** [comment]
Couple of things:

- If you have added an API key in Settings, you can no longer use your login credentials for the `/metrics` endpoint. You must use the generated API keys.
- The API Keys are meant to be used programmatically. But if you still need the browser interface, when the Basic Auth dialog popup you need to copy your API key into the password field, leaving the username field blank.

Feel free to leave more details about what you attempted if you followed these steps and it still doesn't work.

**Comment by qnlbnsl** [comment]
> Changed to `help` as considering a lot of users are already using it without problems.
> 
> Did you try Chrome/Firefox?
I did, no browser worked. 
> Feel free to leave more details about what you attempted if you followed these steps and it still doesn't work.
Yes i read about that as well. So i tried all configurations i could think of. Deleted the API keys and tried login with basic auth, enabled an api key and tried logging in with just the API key as well. Not sure what is going on as i don't see even an attempt to login under the container logs. 
I have the container runnign as a service under a docker swarm where i use a reverse proxy by sharing the network for uptime kuma with my nginx container. I really have no idea as to why the login keeps failing.

**Comment by rezzorix** [comment]
Note that you are also using the HomeAssisstan version; not sure if they have modified/broke something so the metrics arent working as expected.

**Comment by qnlbnsl** [comment]
So an update, i have narrowed down the issue to my data folder. I can only think that the reason is that i enabled API keys once and metrics endpoint never reverted to using basic auth even after deleting the API Key. 
@chakflying  If this is not expected behavior then maybe a better bug would be "disabling API Key does not revert  metrics auth". That being said, it could be something to do with my data folder as it was being run off of a glusterFS drive. I have not removed the old data folder but i did move uptime kuma from my docker swarm to standalone docker.

**Comment by chakflying** [comment]
This is the current intended behavior. Doesn't really explain why you were unable to use the defined API keys.

**Comment by PedroLossolli** [comment]
Hi all,

I ran into the same problem: once an API key is created, HTTP Basic Auth stops working— and deleting the key doesn’t restore it. I traced the root cause to the apiKeysEnabled flag in the settings table of my MySQL backend.

Workaround / Fix

    Connect to your MySQL database.

    Run:

    UPDATE `settings`
    SET `value` = 'false'
    WHERE `key` = 'apiKeysEnabled';

    Restart your Uptime Kuma service.

After this, Basic Auth via username and password works again, even if API keys had previously been enabled.

It looks like deleting an API key does not automatically flip apiKeysEnabled back to false. A built-in option to toggle this flag (or to auto-disable it when all keys are removed) would help avoid manual database edits.
